### PR TITLE
[Search v1] Add sorting 

### PR DESCRIPTION
**Pull request by Kicu**
### Details
 - implements sorting of Search table alongside updating page URL and api calls
 - new arrow icons are taken from Figma
 - I cleaned up types for Routes and navigation - they look much nicer now
 - the added `formatted**` fields are so that we are guaranteed that the values we sort by are the same as values that are actually displayed for the user, so I compute them in the `getSections` function; date needs a bit of special handling because we need different date representation for sorting and different for display

### Fixed Issues

$ https://github.com/Expensify/App/issues/39892
$ https://github.com/Expensify/App/issues/42269
PROPOSAL:

### Tests

1. Open Search view, either via: `Settings > Troubleshoot > New Search Page` (**NOTE:** right now search works only on staging api) or go to `[...]/search/all` URL
2. See the search results appearing in a table
3. click on table headers to toggle between ascending and descending sorting and verify that results are being sorted correctly
4. check in Network tab whether api calls with proper sort params are sent
5. there is no sorting on mobile view at the moment
6. test fix for visual bug: https://github.com/Expensify/App/issues/42269 - see if date and merchant value are aligned together

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/3929868/d5209c60-1621-4cda-8216-87b7a178e868

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/components/Search.tsx
Comment (WojtekBoman) [inline src/components/Search.tsx:72]: Can we create a separate type for `asc | desc` and reuse it in other places where we define `sortOrder`?

**Review by WojtekBoman — APPROVED** [review]

**Comment by Kicu** [comment]
@luacmartins 
I have a question about sorting function:
In the design doc there is an assumption that the sorting function accepts `column` for sorting and does something like `a[columnName] - b[columnName]`.

This is my first approach to this: https://github.com/Expensify/App/pull/42248/files#diff-4e6d1f540181229a6bc7bd2f96266c9b76bc05cf86150ea1e225c657d358aafbR54

However there are 2 problems with this approach:
 - the names of columns do not map to property names (solvable via a mapping object which I did)
 - some of the values that we show in cells are composite values based on more than 1 prop

Examples:
 - `Total` is either `item.modifiedAmount` or `item.amount`
 - `Description` vs `Merchant` are two fields which are interchangeable
 - `From`/`To` can be either: `participant.name`, `participant.displayName` or `participant.login`

Currently I see no other reasonable solution than a chain of `if (columnName === 'from') { // do special user sorting` however a negative of this approach is that the sorting is not really agnostic and working on field, but we are leaking the same display logic we use inside `TransactionItem` into `Search` component.

Do you see any other solution and what way do you want me to move forward?
Second question: are all fields sortable or only some of them?

Feel free to check the draft PR - everything else should be working

**Comment by Kicu** [comment]
@shawnborton in here is the fix for https://github.com/Expensify/App/issues/42269

I added this to "test steps" for the reviewer.

**Comment by shawnborton** [comment]
I feel like when the Search page first loads, we are technically sorting by date right? So as such, I would think we would have this as the starting state:
CleanShot 2024-05-20 at 10 51 33@2x

**Comment by shawnborton** [comment]
For Type and Action, I kinda feel like we don't need a disabled cursor there - let's just use a regular cursor.

Also, some how the header title for the Actions column got messed up - I think that should be centered:
CleanShot 2024-05-20 at 10 53 39@2x

**Comment by shawnborton** [comment]
I don't have any more visual bugs to report (cc @Expensify/design in case you run into anything though), but one thing I did notice was that it doesn't feel like the sorting behavior is actually correct. For instance, if I sort by amount - descending, it seems to be okay but then as I scroll down the list, I see some larger amounts start to appear and it feels like the sorting almost starts over again. Any ideas what's going on there?

**Comment by dannymcclain** [comment]
>I feel like when the Search page first loads, we are technically sorting by date right? So as such, I would think we would have this as the starting state:

Agree!

>For Type and Action, I kinda feel like we don't need a disabled cursor there - let's just use a regular cursor.

I actually think you **should** be able to sort by action and type. For example, I might want to sort by action so that all the expenses with a "Pay" button show in a group. Same with type. I don't see why we wouldn't let you sort by those two.

>but one thing I did notice was that it doesn't feel like the sorting behavior is actually correct.

Same here. I'm noticing this with most of the sortable fields, not just Total.

**Comment by shawnborton** [comment]
> I don't see why we wouldn't let you sort by those two.

Yeah actually that's a totally valid point, I agree with you!

**Comment by dubielzyk-expensify** [comment]
Totally agree with the points you've made Danny. I was initially thinking same as Shawn, but your point is super valid. Good shout!

**Comment by luacmartins** [comment]
>Currently I see no other reasonable solution than a chain of if (columnName === 'from') { // do special user sorting however a negative of this approach is that the sorting is not really agnostic and working on field, but we are leaking the same display logic we use inside TransactionItem into Search component.

Yea, I think we'll have to go with this solution for now. I'm not sure there's a better way to handle this and it's also what we're doing in the backend.

>Second question: are all fields sortable or only some of them?

Most of them should be sortable, I think only exception is the checkbox column (still not present in App).

**Comment by Kicu** [comment]
@shawnborton @dannymcclain @carlosmiceli 

I have allowed sorting by `type` and `action` and centered `Action` header.

Two notes:

### about how sorting feels
I have verified that from the technical point of view the ordering of items is always correct when sorting.
My guess as to why you had the "feeling" as if the items are kinda re-sorting is because backend response is not yet respecting the sorting parameters (?).
In Search view we use lazy loading 50-item chunks, so when you keep scrolling more items are loaded and if api is not yet working then you might get unsorted items and so you have this flash of re-ordering.

@luacmartins is this correct? whats the status of backend?

### Action header
I have centered it now, but it may not look centered to you. That is because I need to reserve the space for the up/down arrow icon. If I don't do this then the whole header text will shift to left after you click on it since the icon will appear and "jump".
For now this is the best middle-ground I can come up with.

<p>


</p>

File: src/components/Search.tsx
Comment (luacmartins) [inline src/components/Search.tsx:41]: I think we should move this to a utils file so we don't clutter the component with a ton of business logic.

File: src/components/Search.tsx
Comment (luacmartins) [inline src/components/Search.tsx:46]:
```suggestion
    const sortingProperty = columnNamesToPropertyMap[sortBy];
```

File: src/components/SelectionList/SearchTableHeader.tsx
Comment (luacmartins) [inline src/components/SelectionList/SearchTableHeader.tsx:25]:
```suggestion
        shouldShow: () => true,
```

File: src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator/BaseCentralPaneNavigator.tsx
Comment (luacmartins) [inline src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator/BaseCentralPaneNavigator.tsx:48]: Let's use a const for `asc`, `desc`

**Review by luacmartins — CHANGES_REQUESTED** [review]
Looking good! I didn't test this yet, but I left a few small comments

File: src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator/BaseCentralPaneNavigator.tsx
Comment (luacmartins) [reply src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator/BaseCentralPaneNavigator.tsx:48]: Also the default here should be `desc`

**Review by luacmartins — COMMENTED** [review]
The headers for total, type and action seem to be misaligned with the cells still 
Screenshot 2024-05-22 at 11 50 25 AM

**Comment by luacmartins** [comment]
Other than my comments above, the feature seems to work really well!

File: src/components/Search.tsx
Comment (luacmartins) [inline src/components/Search.tsx:94]: I wonder if we should be computing a new hash for each set of sorting param since the search results would be different than what we currently have. I think we should at least reset the offset, since the following seems to be possible:

1. Open the search page - the default sort for date desc applies
2. Scroll down to load more results, offset will be updated to 50
3. Sort by a different column, e.g. category
4. We'll fetch the results from the API sorted by category, but we'll get the 2nd "page" since the offset is 50.

Comment (luacmartins) [reply src/components/Search.tsx:94]: Maybe one optimization would be when `hasMoreResults = false` since we know we have all available data and wouldn't need to fetch anything else to properly sort things in the client.

**Comment by luacmartins** [comment]
>I have centered it now, but it may not look centered to you. That is because I need to reserve the space for the up/down arrow icon. If I don't do this then the whole header text will shift to left after you click on it since the icon will appear and "jump".
For now this is the best middle-ground I can come up with.

@Kicu would adding a min-width to these prevent the shift when the arrow is shown?

File: src/components/Search.tsx
Comment (Kicu) [reply src/components/Search.tsx:94]: > I wonder if we should be computing a new hash for each set of sorting param

I agree with this and I will add sortBy and sortOrder to hash generation.

However I think the scenario you described would not happen because in this specific line offset is set always to `0` inside this use effect.
So in your scenario in point 3 if you sort by different column or change order, then this `useEffect` block will be in place and a fresh call will be made with `offset: 0`. If you keep scrolling then another call will be made with offset 50 or whatever, but the one with the 0 was already there.

Perhaps another solution is to scroll list to top if you switch sort/order?

File: src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator/BaseCentralPaneNavigator.tsx
Comment (Kicu) [reply src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator/BaseCentralPaneNavigator.tsx:48]: Should I make the change where it is always `desc` first when you press on a header and `asc` on second press?

Comment (luacmartins) [reply src/libs/Navigation/AppNavigator/Navigators/CentralPaneNavigator/BaseCentralPaneNavigator.tsx:48]: Yes please

**Comment by Kicu** [comment]
@luacmartins 
pushed all of your suggestions.
I needed to do some extra tricks for `Search` results, because adding sorting props to `hash` generated different Onyx key and every click on table header would reload the data, flash the skeleton view and it was very bad UX.
So now we're saving last fetched data and use it if current data onyx is missing - the experience is much nicer.

**About the width + centering the header**
> would adding a min-width to these prevent the shift when the arrow is shown?

min width wouldn't really change much, because the problem is not with the code/CSS but rather its more basic: we have 2 ways of doing this
A. if there is no sorting and no icon -> text will be right/center aligned but it will jump when icon appears
B. if there is no sorting but still empty space for icon -> text will not look aligned but it will NOT jump when icon appears

#### behavior A
https://github.com/Expensify/App/assets/3929868/b4b74d16-2831-4d94-99e3-50609f7c5bef

#### behavior B (current)
https://github.com/Expensify/App/assets/3929868/b576885c-357f-4f10-85c1-ffc53ccd11c8


I guess a different solution that I see would be to always have some kind "default" icon that would be always shown to the right, like the up-down arrow and then the whole thing would be always centered and not jump.
👆 about last point @dannymcclain @shawnborton

**Comment by dubielzyk-expensify** [comment]
Yeah, I'm aligned that option A is probably the right call here. It doesn't bother me much either though I do think we might wanna make the Type column a tiny bit larger so that it doesn't look as cramped with the arrow in the header

**Comment by Kicu** [comment]
Option A brought back and I made the `type` tab a bit wider.

https://github.com/Expensify/App/assets/3929868/5ac61343-8f91-4650-bd31-a406af469476

**Comment by shawnborton** [comment]
Yup, I'm aligned on that too - Option A doesn't feel bad at all and I agree it does feel like what I would expect to see in this scenario.

File: src/libs/SearchUtils.ts
Comment (c3024) [inline src/libs/SearchUtils.ts:64]:
```suggestion
            const formattedTo = to?.name ?? to?.displayName ?? to?.login ?? '';
```

https://github.com/Expensify/App/assets/102477862/e5e48a9c-90c3-439e-9946-3d2e1b364ba0
Screenshot 2024-05-27 at 9 38 31 PM

`to` is `undefined` for some in my high traffic account.

**Comment by c3024** [comment]
Merchant field sorting has issues.

1. Merchant field is not displayed for some transactions
2. Distance is shown as merchant
3. Not sorted correctly


https://github.com/Expensify/App/assets/102477862/51fa685e-07d6-4f43-b6f4-986c80bd180b

**Review by luacmartins — DISMISSED** [review]
Changes LGTM and tests well. I think we need to address a few things in a follow up though, mostly regarding the shown columns and their widths which can change based on sorting and makes things look a bit broken. Maybe it's best to just show a loading indicator when we don't have data, but I'll leave that to a follow up.

https://github.com/Expensify/App/assets/22219519/dc400fa8-4164-4061-96a2-c32064bd0c0e

**Comment by luacmartins** [comment]
@c3024 let's not block this PR on [this suggestion](https://github.com/Expensify/App/pull/42248#discussion_r1616234829). Could you please continue the review?

**Comment by c3024** [comment]
Sorting works well for the first pages but after some pages, the sorting is not in correct descending order for desc for `Total`.


https://github.com/Expensify/App/assets/102477862/b29dc01c-92c6-43c6-be09-10800302747a

File: src/libs/SearchUtils.ts
Comment (Kicu) [reply src/libs/SearchUtils.ts]: fixed

**Comment by Kicu** [comment]
@c3024 thanks for the review, some comments:

> Merchant field is not displayed for some transactions

that is based on api data and decision made by @luacmartins (https://github.com/Expensify/App/blob/main/src/components/SelectionList/TransactionListItem.tsx#L191)

> Distance is shown as merchant

this is based on data that I get from backend as well, frontend displays data:
merchant-distance

> Not sorted correctly (same for `to` field as well)

 - fixed sorting for merchant field (but the empty fields are still displayed)

About sorting by `to` and `total` this is not something that I understand and I do not have the means to reproduce it.
Carlos is there any way to get an account that would have hundreds of transaction items? I'm looking at the logic for sorting by `total` and by `to` and I really see nothing wrong with the code. I have no idea why suddenly the sorting breaks.

I would like to see the api responses that are made on lazy loading. Not sure how to move forward with this. 
I will try to merge main and fix conflicts however.

**Comment by Kicu** [comment]
**updated sort by merchant**

https://github.com/Expensify/App/assets/3929868/f82e2033-d97c-4437-b767-27e44172d495

**Comment by Kicu** [comment]
guys, I will not be merging and solving conflicts on this yet, as this PR https://github.com/Expensify/App/pull/42410 also introduces big changes to `TransactionListItem` so I will wait for this one to be merged, and then follow up with sorting.

Meanwhile we can still investigate if sorting logic works correctly

**Comment by c3024** [comment]
I just checked again. `To` and `Total` sorts work well now.


https://github.com/Expensify/App/assets/102477862/fc68ef22-0502-46cf-a619-915173efcc81


However, (unrelated to this) when I scroll back up the app crashed. There are some undefined `from` as well apparently.


Screenshot 2024-05-28 at 5 57 01 PM

**Comment by luacmartins** [comment]
@Kicu I merged the `ReportListItem` PR, can you resolve conflicts now please?

**Comment by c3024** [comment]
There is a require cycle with PR #42410

> WARN  Require cycle: src/libs/SearchUtils.ts -> src/components/SelectionList/Search/ReportListItem.tsx -> src/libs/SearchUtils.ts

We might fix that as well here.

**Comment by Kicu** [comment]
@luacmartins @c3024 
This is ready for re-review. Unfortunately a lot has changed when I was resolving the conflicting files and so there is some new code.
Every commit before merges is unchanged - but the last 3 commits are new.

The biggest changes are:
 - getSortedSections is now a generic function that handles specific types
 - report items do not have any sorting yet
 - pieces of code are reused between components


@luacmartins for now the `Receipt` column is not sortable because I wasn't sure by which field to do the sorting.
If you can tell me we can add it, to do so:
 - in `SearchTableHeader` remove the `isSortable: false` flag
 - in `SearchUtils` update the mapping: `columnNamesToSortingProperty`, introduce new property if needed

Please review :)

**Comment by luacmartins** [comment]
Thanks @Kicu! Not sure if we wanna sort by receipts or not, but we can definitely go ahead without sorting that for now. As for reports, we landed on no sorting for now, so all good there.

**Comment by luacmartins** [comment]
LGTM [this](https://github.com/Expensify/App/pull/42248#pullrequestreview-2081281412) is still a thing that we should address in a follow up

**Review by luacmartins — APPROVED** [review]
LGTM

**Comment by luacmartins** [comment]
@c3024 can you review again please?

**Comment by c3024** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by c3024 — APPROVED** [review]
LGTM!

**Comment by c3024** [comment]
There are cases of `undefined` for `from` as well and it crashed on scrolling up accessing `displayName` on `from`.  Apart from that, everything looks good.

**Review by luacmartins — APPROVED** [review]

File: src/libs/SearchUtils.ts
Comment (s77rt) [inline src/libs/SearchUtils.ts:28]: The description can be used as sort column (https://github.com/Expensify/App/issues/44431)

File: src/components/Search.tsx
Comment (SzymczakJ) [inline src/components/Search.tsx:49]: ref comment

**Pull request merged**
